packages/grid/_modules_/grid/hooks/features/rows/useGetRowId.ts

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import { ApiRef } from '../../../models/api/apiRef';
+import { RowData, RowIdGetter } from '../../../models/rows';
+import { useApiMethod } from '../../root/useApiMethod';
+
+const getDefaultRowId = (row: RowData) => row.id;
+
+export const useGetRowId = (apiRef: ApiRef, getRowIdProp?: RowIdGetter) => {
+  const getRowId = React.useCallback(
+    (row: RowData) => {
+      if (!getRowIdProp) {
+        return getDefaultRowId(row);
+      }
+      return getRowIdProp(row);
+    },
+    [getRowIdProp],
+  );
```

Comment:
We can still inverse the branching; reduce the amount of work done when processing 100,000 rows.